smallrye-reactive-messaging-kafka/src/test/java/io/smallrye/reactive/messaging/kafka/commit/KafkaRedisCheckpointCommit.java

```diff
@@ -0,0 +1,112 @@
+package io.smallrye.reactive.messaging.kafka.commit;
+
+import java.util.Arrays;
+import java.util.Optional;
+import java.util.function.BiConsumer;
+import java.util.function.Function;
+
+import javax.enterprise.context.ApplicationScoped;
+
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.common.TopicPartition;
+
+import io.smallrye.common.annotation.Identifier;
+import io.smallrye.mutiny.Uni;
+import io.smallrye.reactive.messaging.kafka.KafkaConnectorIncomingConfiguration;
+import io.smallrye.reactive.messaging.kafka.KafkaConsumer;
+import io.vertx.core.json.Json;
+import io.vertx.mutiny.core.Vertx;
+import io.vertx.mutiny.core.buffer.Buffer;
+import io.vertx.mutiny.redis.client.Redis;
+import io.vertx.mutiny.redis.client.RedisAPI;
+import io.vertx.mutiny.redis.client.Response;
+import io.vertx.redis.client.RedisOptions;
+
+/**
+ * This implementation is probably wrong.
+ */
+public class KafkaRedisCheckpointCommit extends KafkaCheckpointCommit {
+
+    private final Redis redis;
+
+    private RedisAPI redisAPI;
```

Comment:
Do not use the RedisAPI, use the Redis object directly and send commands. I got bitten by the Redis API.

Reply:
👍